src/widget/wpushbutton.cpp

```diff
@@ -337,23 +355,43 @@ void WPushButton::paintEvent(QPaintEvent* e) {
 
     QString text = m_text.at(idx);
     if (!text.isEmpty()) {
-        p.drawText(rect(), m_align.at(idx), text);
+        QFontMetrics metrics(font());
+        // ToDo(ronso0) Consider css padding for buttons with border images
+        // * read QWidget::style()->property("padding-left");
+        // * adjust width()
+        // * transform rect()
+//        int textWidth = width() - lPad - rPad;
+//        QRect textRect = rect().adjust(x1, y1, x2, y2);
+        QString elidedText = metrics.elidedText(text, m_elideMode, width());
+        p.drawText(rect(), m_align.at(idx), elidedText);
     }
 }
 
+void WPushButton::enterEvent(QEvent *event) {
+    m_bHovered = true;
+    restyleAndRepaint();
+    return QWidget::enterEvent(event);
+}
+
+void WPushButton::leaveEvent(QEvent *event) {
+    m_bHovered = false;
+    restyleAndRepaint();
+    return QWidget::leaveEvent(event);
+}
+
 void WPushButton::mousePressEvent(QMouseEvent * e) {
     const bool leftClick = e->button() == Qt::LeftButton;
     const bool rightClick = e->button() == Qt::RightButton;
 
     if (m_leftButtonMode == ControlPushButton::POWERWINDOW
             && m_iNoStates == 2) {
         if (leftClick) {
-            m_clickTimer.setSingleShot(true);
-            m_clickTimer.start(ControlPushButtonBehavior::kPowerWindowTimeMillis);
-
-            double emitValue = getControlParameterLeft() == 0.0 ? 1.0 : 0.0;
-            setControlParameterLeftDown(emitValue);
+            if (getControlParameterLeft() == 0.0) {
```

Comment:
Is this related to this PR? What does it solve?

Reply:
oh, not intended. this must have slipped in from an unrelated branch.
I'll revert it.